### PR TITLE
Cor 340 react native tests

### PR DESCRIPTION
**Pull request by nilueps**
Sorry about the gigantic PR. Setting up eslint was one of the things I tried to get to the bottom of some nagging errors with jest and I ended up auto-fixing the entire frontend dir.

### CODE REVIEW

File: web/app/frontend/__tests__/utils/storage.test.js [hunk 1]
```diff
@@ -1,17 +1,73 @@
...
+SecureStore.getItemAsync.mockResolvedValue(key);
+CryptoJS.AES.decrypt.mockResolvedValue(JSON.stringify(data));
+let encryptedData;
+(async () => {
+    encryptedData = (
+        await CryptoJS.AES.encrypt(JSON.stringify(data), key)
+    ).toString();
+    CryptoJS.AES.encrypt.mockResolvedValue(encryptedData);
+    AsyncStorage.getItem.mockResolvedValue(encryptedData);
+})();
```
Comment (internetti) [inline web/app/frontend/__tests__/utils/storage.test.js:23]: this stuff is usually done within the describe, I think it would be cleaner that way. 
then you also wouldn't have to do this closure to prepare the encryptedData

File: web/app/frontend/jest.setup.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import 'react-native-gesture-handler/jestSetup';
+import mockAsyncStorage from '@react-native-async-storage/async-storage/jest/async-storage-mock';
+import 'core-js-api-client';
+
+jest.mock('@react-native-async-storage/async-storage', () => mockAsyncStorage);
```
Comment (internetti) [inline web/app/frontend/jest.setup.js:5]: is there a reason there are two files with this? or can one of them be removed?

File: web/app/frontend/src/components/Router.tsx [hunk 3]
```diff
@@ -33,63 +31,79 @@ export default function Router() {
...
+                            initialParams={{ formId: '', databaseId: '' }}
                             options={{
-                                title: routes.records.title
+                                title: routes.records.title,
                             }}
                         >
-                            {(props) =>
-                                <RecordsScreen {...props} state={state} dispatch={dispatch}/>
-                            }
+                            {({ navigation, route }) => (
```
Comment (internetti) [inline web/app/frontend/src/components/Router.tsx:65]: why did you only do this change for this screen?

File: web/app/frontend/src/components/screens/FormsScreen.tsx [hunk 4]
```diff
@@ -1,22 +1,24 @@
...
-import {FlatList, Text, TouchableOpacity, View} from 'react-native';
-import useApiClient from "../../utils/clients";
-import routes from "../../constants/routes";
-import {FormDefinition} from "core-js-api-client/lib/types/types";
+import testIds from '../../testIds';
+import {FormsScreenProps} from '../../types';
+import {useApiClient} from '../../utils/useApiClient';
 
-const FormsScreen: React.FC<any> = ({navigation}) => {
+const FormsScreen = ({navigation}: FormsScreenProps) => {
```
Comment (internetti) [inline web/app/frontend/src/components/screens/FormsScreen.tsx:12]: this change has not been done consistently, can we pick one version?

File: web/app/frontend/src/types.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import { StackScreenProps } from '@react-navigation/stack';
```
Comment (internetti) [inline web/app/frontend/src/types.tsx:1]: can we move this to the types folder?
and I think this doesn't need to be a tsx file, ts should be enough

File: web/app/frontend/tsconfig.json [hunk 6]
```diff
@@ -1,7 +1,15 @@
 {
   "extends": "expo/tsconfig.base",
   "compilerOptions": {
-    "jsx": "react-native",
-    "strict": true
-  }
+    "strict": true,
```
Comment (internetti) [inline web/app/frontend/tsconfig.json:4]: were these changes necessary for the tests?

**Review by internetti — CHANGES_REQUESTED** [review]
how do I do the linting, is there a command? don't see anything in the package json

File: web/app/client/src/types/types.ts [hunk 7]
```diff
@@ -80,6 +80,15 @@ export class FormDefinition {
     public folderId: string = ""
     public name: string = ""
     public fields: FieldDefinition[] = []
+
+    constructor(id = "", code = "", databaseId = "", folderId = "", name = "", fields = []) {
```
[the remark this answers is not in the document]
Comment (nilueps) [reply web/app/client/src/types/types.ts:84]: I don't remember tbh, but I'm guessing ts was throwing type errors and I wanted a constructor :/

File: web/app/frontend/src/testIds.js [hunk 8]
```diff
@@ -0,0 +1,3 @@
+export default {
```
[the remark this answers is not in the document]
Comment (nilueps) [reply web/app/frontend/src/testIds.js:1]: moved to constants dir

File: web/app/frontend/jest.setup.js (hunk 2 above)
Comment (nilueps) [reply web/app/frontend/jest.setup.js:5]: not sure what you mean

Comment (internetti) [reply web/app/frontend/jest.setup.js:5]: I mean this file: web/app/frontend/__mocks__/@react-native-async-storage/async-storage.js
it basically has this exact line as content, and I'm assuming one can go?

**Comment by nilueps** [comment]
given the XXL nature of this PR; I suggest we merge it ASAP and make tickets for test requirements
